File: study_class.php
<?php
class TestClass{
  private $set_num;

  public function __construct(){
    $this->set_num = array();
  }
  public function add($num){
    array_push($this->set_num, $num);
  }
  public function remove(){
    array_pop($this->set_num);
  }
  public function addition(){
    return array_sum($this->set_num)."\n";
  }
  public function kakezan(){
    return array_product($this->set_num)."\n";
  }
}

File: study_class.rb
class TestClass
  attr_accessor :set_num
  def initialize
    @set_num = []
  end
  def add(num)
    @set_num.push(num)
  end
  def remove
    @set_num.pop
  end
  def addition
    @set_num.inject(:+)
  end
  def kakezan
    @set_num.inject(:*)
  end
end
=begin

phpバージョン
array_sum($this->set_num);

rubyバージョン
@set_num.inject(:*)
=end
